pkg/controllers/proxy/network_services_controller.go

```diff
@@ -793,6 +793,42 @@ func (nsc *NetworkServicesController) publishMetrics(serviceInfoMap serviceInfoM
 	return nil
 }
 
+func epCompatible(a, b []endpointsInfo) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for _, epA := range a {
+		for _, epB := range b {
+			if epA == epB {
+				return true
```

Comment:
This would stop after the first match without considering the others right?

i.e. say `a` contains `["foo","bar"]` endpoints and `b` contains `["foo", "baz"]`. The first iteration of this nested loop would see `"foo" == "foo"` and then return true, without considering that `"bar" != "baz"`

You should invert the condition here so that if ever `epA != epB` then `return false` otherwise if you make it all the way to the end of the function `return true`

Reply:
Absolutely true, that logic mistake probably happened as I split out the endpoint comparison to a separate method - same as the missing early exit on that loop.